From 1d0bc2e9fbbfca05624e35cb16161b4121724d2f Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Tue, 26 Nov 2024 21:32:50 +0100
Subject: [PATCH] [doged] Rename dogecoin* executables to doge* (#85)

This is part of an effort to rebrand the node from Dogecash to doged, see #82.

Current-ABC adaptation (2026-10-14): Apply final daemon and CLI identities directly to current CMake targets, help and functional harness; preserve current build APIs, source filenames and BITCOIND/BITCOINCLI overrides. Incorporates daemon/CLI rename intent from f48fe80351b6c992c6a9b239e6442a2a9af506a7 and a1a57dea066c6efa270c9ba2601a93340af38f23. Qt and auxiliary tools retain upstream target names pending separate packaging review.

Original-Commit: 75dc738ffe920e6b99690fd8206071dda9c357cb

diff --git a/doc/man/CMakeLists.txt b/doc/man/CMakeLists.txt
index 4a7c0341df07cc5bb3f20ef49c88c340e1bc2dbd..e2fe32c4ca65545f5b442e6c28024bc3ac03be88 100644
--- a/doc/man/CMakeLists.txt
+++ b/doc/man/CMakeLists.txt
@@ -18,11 +18,11 @@ function(generate_manpages)
 				"${CMAKE_SOURCE_DIR}/cmake/utils/log-and-print-on-failure.sh"
 				"${CMAKE_BINARY_DIR}/doc/man/gen-manpage-${TARGET}.log"
 				"${CMAKE_SOURCE_DIR}/doc/man/gen-manpages.sh"
-				"$<TARGET_FILE:bitcoind>"
+				"$<TARGET_FILE:doged>"
 				"$<TARGET_FILE:${TARGET}>"
 				"${MAN_PAGE}"
 			DEPENDS
-				bitcoind
+				doged
 				"${TARGET}"
 		)
 		add_custom_target(doc-manpage-${TARGET}
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 25fee230af2993ec0a3c63e51aef76d1c3aa233d..9f42dcce5c0c27fc07469d97a69579f1cb29f887 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (c) 2017 The Bitcoin developers
 
-project(bitcoind)
+project(doged)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -16,7 +16,7 @@ endforeach()
 
 option(BUILD_WALLET "Activate the wallet functionality" ON)
 option(BUILD_ZMQ "Activate the ZeroMQ functionalities" ON)
-option(BUILD_CLI "Build bitcoin-cli" ON)
+option(BUILD_CLI "Build doge-cli" ON)
 option(BUILD_TX "Build bitcoin-tx" ON)
 option(BUILD_QT "Build bitcoin-qt" ON)
 option(BUILD_SEEDER "Build bitcoin-seeder" ON)
@@ -789,21 +789,21 @@ if(BUILD_SEEDER)
 	add_subdirectory(seeder)
 endif()
 
-# bitcoin-cli
+# doge-cli
 if(BUILD_CLI)
-	add_executable(bitcoin-cli bitcoin-cli.cpp)
-	generate_windows_version_info(bitcoin-cli
+	add_executable(doge-cli bitcoin-cli.cpp)
+	generate_windows_version_info(doge-cli
 		DESCRIPTION "JSON-RPC client for ${PACKAGE_NAME}"
 	)
 
-	target_link_libraries(bitcoin-cli common rpcclient)
-	link_event(bitcoin-cli event)
+	target_link_libraries(doge-cli common rpcclient)
+	link_event(doge-cli event)
 
-	add_to_symbols_check(bitcoin-cli)
-	add_to_security_check(bitcoin-cli)
+	add_to_symbols_check(doge-cli)
+	add_to_security_check(doge-cli)
 
-	install_target(bitcoin-cli)
-	install_manpages(bitcoin-cli)
+	install_target(doge-cli)
+	install_manpages(doge-cli)
 endif()
 
 # bitcoin-tx
@@ -994,17 +994,17 @@ if(BUILD_CHAINSTATE)
 	add_to_security_check(bitcoin-chainstate)
 endif()
 
-# bitcoind
-add_executable(bitcoind bitcoind.cpp)
-target_link_libraries(bitcoind server)
-generate_windows_version_info(bitcoind
+# doged
+add_executable(doged bitcoind.cpp)
+target_link_libraries(doged server)
+generate_windows_version_info(doged
 	DESCRIPTION "Bitcoin node with a JSON-RPC server"
 )
-add_to_symbols_check(bitcoind)
-add_to_security_check(bitcoind)
+add_to_symbols_check(doged)
+add_to_security_check(doged)
 
-install_target(bitcoind)
-install_manpages(bitcoind)
+install_target(doged)
+install_manpages(doged)
 
 # Bitcoin-qt
 if(BUILD_QT)
diff --git a/src/bitcoin-cli.cpp b/src/bitcoin-cli.cpp
index 91cd1918995959e33c4bdf43e187a93d1e5f0660..ec57b1df9f55bdc10ee548f3cd28323b09abbdc7 100644
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -82,7 +82,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
             "arguments are number of blocks to generate (default: %s) and "
             "maximum iterations to try (default: %s), equivalent to RPC "
             "generatetoaddress nblocks and maxtries arguments. Example: "
-            "bitcoin-cli -generate 4 1000",
+            "doge-cli -generate 4 1000",
             DEFAULT_NBLOCKS, DEFAULT_MAX_TRIES),
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg(
@@ -160,7 +160,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
     argsman.AddArg(
         "-rpcwallet=<walletname>",
         "Send RPC for non-default wallet on RPC server (needs to exactly match "
-        "corresponding -wallet option passed to bitcoind). This changes the "
+        "corresponding -wallet option passed to doged). This changes the "
         "RPC endpoint used, e.g. http://127.0.0.1:8332/wallet/<walletname>",
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
 }
@@ -211,14 +211,14 @@ static int AppInitRPC(int argc, char *argv[]) {
             strUsage += FormatParagraph(LicenseInfo());
         } else {
             strUsage += "\n"
-                        "Usage:  bitcoin-cli [options] <command> [params]  "
+                        "Usage:  doge-cli [options] <command> [params]  "
                         "Send command to " PACKAGE_NAME "\n"
-                        "or:     bitcoin-cli [options] -named <command> "
+                        "or:     doge-cli [options] -named <command> "
                         "[name=value]...  Send command to " PACKAGE_NAME
                         " (with named arguments)\n"
-                        "or:     bitcoin-cli [options] help                "
+                        "or:     doge-cli [options] help                "
                         "List commands\n"
-                        "or:     bitcoin-cli [options] help <command>      Get "
+                        "or:     doge-cli [options] help <command>      Get "
                         "help for a command\n";
 
             strUsage += "\n" + gArgs.GetHelpMessage();
@@ -510,7 +510,7 @@ public:
 
         const UniValue &networkinfo{batch[ID_NETWORKINFO]["result"]};
         if (networkinfo["version"].getInt<int>() < 230000) {
-            throw std::runtime_error("-netinfo requires bitcoind server to be "
+            throw std::runtime_error("-netinfo requires doged server to be "
                                      "running v0.23.0 and up");
         }
 
@@ -813,7 +813,7 @@ static UniValue CallRPC(BaseRequestHandler *rh, const std::string &strMethod,
         }
         throw CConnectionFailed(
             strprintf("Could not connect to the server %s:%d%s\n\nMake sure "
-                      "the bitcoind server is running and that you are "
+                      "the doged server is running and that you are "
                       "connecting to the correct RPC port.",
                       host, port, responseErrorMessage));
     } else if (response.status == HTTP_UNAUTHORIZED) {
@@ -919,7 +919,7 @@ static void ParseError(const UniValue &error, std::string &strPrint,
         if (err_code.isNum() &&
             err_code.getInt<int>() == RPC_WALLET_NOT_SPECIFIED) {
             strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to "
-                        "bitcoin-cli command line.";
+                        "doge-cli command line.";
         }
     } else {
         strPrint = "error: " + error.write();
diff --git a/src/bitcoind.cpp b/src/bitcoind.cpp
index 16d1201338e189f0927486a505e5669876663aff..310ba7a6b1ad7f9586cfa7042c1e5499d5a59301 100644
--- a/src/bitcoind.cpp
+++ b/src/bitcoind.cpp
@@ -162,7 +162,7 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
         if (args.IsArgSet("-version")) {
             strUsage += FormatParagraph(LicenseInfo()) + "\n";
         } else {
-            strUsage += "\nUsage:  bitcoind [options]                     "
+            strUsage += "\nUsage:  doged [options]                     "
                         "Start " PACKAGE_NAME "\n";
             strUsage += "\n" + args.GetHelpMessage();
         }
@@ -201,12 +201,12 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
             if (!IsSwitchChar(argv[i][0])) {
                 return InitError(Untranslated(
                     strprintf("Command line contains unexpected token '%s', "
-                              "see bitcoind -h for a list of options.",
+                              "see doged -h for a list of options.",
                               argv[i])));
             }
         }
 
-        // -server defaults to true for bitcoind but not for the GUI so do this
+        // -server defaults to true for doged but not for the GUI so do this
         // here
         args.SoftSetBoolArg("-server", true);
         // Set this early so that parameter interactions go to console
@@ -311,7 +311,7 @@ int main(int argc, char *argv[]) {
 
     SetupEnvironment();
 
-    // Connect bitcoind signal handlers
+    // Connect doged signal handlers
     noui_connect();
 
     return (AppInit(node, argc, argv) ? node.exit_status.load() : EXIT_FAILURE);
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index c11a6959c9673009ca266507bbf5ad7c3913bf69..7ee01d728ff0736eded752d4c7b0a612cdb26669 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,7 +26,7 @@ include(TestSuite)
 
 set(_TEST_TARGET_DEPENDS "")
 if(BUILD_CLI)
-	list(APPEND _TEST_TARGET_DEPENDS bitcoin-cli)
+	list(APPEND _TEST_TARGET_DEPENDS doge-cli)
 endif()
 if(BUILD_WALLET)
 	list(APPEND _TEST_TARGET_DEPENDS bitcoin-wallet)
@@ -73,7 +73,7 @@ function(add_functional_test_check TARGET DESCRIPTION)
 		CUSTOM_TARGET_ARGS
 			COMMENT "Running ${DESCRIPTION}"
 			DEPENDS
-				bitcoind
+				doged
 				${_TEST_TARGET_DEPENDS}
 				${CMAKE_CURRENT_BINARY_DIR}/functional/test_runner.py
 			USES_TERMINAL
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 1f2521c62fff59087128bd30b53049d37751d8a5..803f974a07dbfb55f1a0f7a0080136b9849291de 100644
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -334,12 +334,12 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
         fname_bitcoind = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoind{config['environment']['EXEEXT']}",
+            f"doged{config['environment']['EXEEXT']}",
         )
         fname_bitcoincli = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoin-cli{config['environment']['EXEEXT']}",
+            f"doge-cli{config['environment']['EXEEXT']}",
         )
         self.options.bitcoind = os.getenv("BITCOIND", default=fname_bitcoind)
         self.options.bitcoincli = os.getenv("BITCOINCLI", default=fname_bitcoincli)
